Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -1271,30 +1277,50 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
             return;
         }
 
-        var treeOutline = this.treeOutline;
-        var wasExpanded = this.expanded;
+        let treeOutline = this.treeOutline;
+        let wasExpanded = this.expanded;
+        let [newTagName, ...attributes] = newText.trim().split(" ");
 
-        function changeTagNameCallback(error, nodeId)
-        {
+        let changeTagNameCallback = (error, nodeId) => {
             if (error || !nodeId) {
                 cancel();
                 return;
             }
 
-            var node = WI.domManager.nodeForId(nodeId);
+            let node = WI.domManager.nodeForId(nodeId);
+
+            if (!attributes.length) {
+                nodeUpdatedCallback(node, {moveFocus: true});
+                return;
+            }
+
+            let attributesAddedCallback = (error, nodeId) => {
+                if (error) {
+                    cancel();
+                    return;
+                }
+
+                nodeUpdatedCallback(node);
```

Comment:
in the video it looks like if you hit enter with a bunch of attributes we just completely stop editing.  do we not want to focus the first attribute just in case the developer mistypes?  IMO that was kinda odd

Reply:
The reasoning was that if you just typed in attributes, you don't need to go edit them straight away.

Focusing the first attribute is not guaranteed to focus what the author just typed in. Attributes are sorted alphabetically. We'd have to identify which was the first attribute in the string and switch that into edit mode. 
IMO, this is unnecessary. If there's a need to edit, the author can easily move into that mode.

Reply:
i suppose.  i kinda feel like we should at least focus _something_ as otherwise the developer has to take an extra step to re-select in order to navigate using tab (especially for things like screen readers), but i think we have similar behavior with some other interactions (e.g. contextmenu > Add > Attribute) so im fine with leaving this for now

Reply:
> i kinda feel like we should at least focus something as otherwise the developer has to take an extra step to re-select in order to navigate using tab

That's a good argument. 👍  I lived on this for a bit and I agree that not moving focus makes tab navigation awkward. 

I added it back and changed it to select the first attribute if we just added attributes with the tag name (instead of creating a new empty attribute when committing just a tag name) 

Also added a drive-by fix to the automatic selection of attribute names when they're value-less (ex: `disabled`). Before, it caused an unhandled exception when tabbing through attributes because it tries to select the attribute value.

https://github.com/WebKit/WebKit/blob/76bd9ad78d4e916b082bd10c8346d287466216c7/Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js#L1018